local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -208,40 +208,31 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
   pose_to_wp *= planner_info_.min_speed;
 
   output_.goto_position = toPoint(toEigen(pose_.pose.position) + pose_to_wp);
+  output_.adapted_goto_position = output_.goto_position;
+  output_.smoothed_goto_position = output_.goto_position;
+  smoothed_goto_location_ = toEigen(output_.smoothed_goto_position);
 }
 
 void WaypointGenerator::smoothWaypoint(double dt) {
-  Eigen::Vector2f vel_cur_xy(curr_vel_.twist.linear.x,
-                             curr_vel_.twist.linear.y);
-  Eigen::Vector2f vel_sp_xy((output_.adapted_goto_position.x -
-                             last_position_waypoint_.pose.position.x) /
-                                dt,
-                            (output_.adapted_goto_position.y -
-                             last_position_waypoint_.pose.position.y) /
-                                dt);
-
-  Eigen::Vector2f accel_diff = (vel_sp_xy - vel_cur_xy) / dt;
-  Eigen::Vector2f accel_cur = (vel_cur_xy - last_velocity_) / dt;
-  Eigen::Vector2f jerk_diff = (accel_diff - accel_cur) / dt;
-  float max_jerk = max_jerk_limit_param_;
-
-  // velocity-dependent max jerk
-  if (min_jerk_limit_param_ > 0.001f) {
-    max_jerk *= vel_cur_xy.norm();
-    if (max_jerk < min_jerk_limit_param_) max_jerk = min_jerk_limit_param_;
-  }
-
-  if (jerk_diff.squaredNorm() > max_jerk * max_jerk && max_jerk > 0.001f) {
-    jerk_diff = max_jerk * jerk_diff.normalized();
-    vel_sp_xy = (jerk_diff * dt + accel_cur) * dt + vel_cur_xy;
+  if (smoothing_speed_ > 0.01) {
```

Comment:
Maybe it's worth noting that this smoothing only works for position setpoints. At first sight, I thought it was also smoothing when using the velocity setpoints... I got confused by the name `smoothed_goto_velocity`. Can we add that function and clarifying comments, should be quite similar, no?

Also, I just realized that I don't understand the `struct waypointResult` altogether. there is a `position_waypoint`, `goto_position`, `adapted_goto_position` and `smoothed_goto_position`. Desperately needs some comments in the code, those all sound the same to me.

Reply:
Here's a log of the same mission using velocity setpoints, it still seems to be smoothed, but travels a lot faster.
https://logs.px4.io/plot_app?log=d7ae64d4-c6d6-42f1-a60c-1f5a776c595f

Reply:
I will rename the variable to make it more clear